CMakePresets.json

```diff
@@ -0,0 +1,407 @@
+{
+    "version": 3,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 22,
+        "patch": 0
+    },
+    "configurePresets": [
+        {
+            "name": "develop",
+            "description": "enable tests and examples",
+            "hidden": true,
+            "binaryDir": "${sourceDir}/build/${presetName}",
+            "cacheVariables": {
+                "alpaka_BUILD_EXAMPLES": {
+                    "type": "BOOL",
+                    "value": "ON"
+                },
+                "BUILD_TESTING": {
+                    "type": "BOOL",
+                    "value": "ON"
+                }
+            }
+        },
+        {
+            "name": "cpu-serial-gcc",
+            "description": "use g++ compiler with serial back-end",
+            "inherits": "develop",
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER": {
+                    "type": "STRING",
+                    "value": "g++"
+                },
```

Comment:
I assume this would break my workflow. I configure with:
`CXX=g++-13 cmake ..`. For clang, I do not even have a `clang++` installed, only `clang++-version`.

Reply:
You version would hard work together with IDEs such like Visual Studio Code. In special for vsc on remote systems, it is pretty hard to configure environment variables.

Reply:
For CLion, this was easy to configure in the Toolchain menu. But anyhow, I just made a suggestion. I don't need the preset file here, so if you want to hardcode `g++` than I don't mind.

Reply:
The basic idea was, that you can easily overwrite the parameter with `cmake --preset cmake --preset cpu-serial-gcc -DCMAKE_CXX_COMPILER=g++-13`. But we can also remove the gcc and clang version and provides only one configuration per cpu back-end.

Reply:
Now, I only set the generic device compiler (`clang++`, `hipcc`, `ipcx`) where it is required.